### ISSUE TITLE
Switch from bors to GH merge queues

### ISSUE THREAD

**Issue by oleschoenburg** [camunda/zeebe#12097]
Now that [Merge Queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) is in public beta, we could switch over to that and away from our own bors instance.

To ensure compatibility with our existing workflows we should test this out in a fork first to ensure a smooth transition.

Some notes:
- The CI workflow needs a new [trigger](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions). 
- This enables https://github.com/camunda/zeebe/issues/10588

**Comment by korthout** [comment]
ZPA triage: Sounds like a great improvement, but not a priority for us right now. `Priority: later`.

**Comment by megglos** [comment]
ZDP planning: we could enable it in parallel to bors for evaluation, would bring the benefit of being able to parallelize merges for different target branches (we have one more stable branch now), let's time-box 1h

**Comment by megglos** [comment]
ZDP-Planning:
- not urgent right now and no capacity, pushing back to backlog

**Comment by nicpuppa** [comment]
FYI, `bors` project is officially [deprecated](https://bors.tech/newsletter/2023/05/01/tmib-76/). And Merge Queue is now [generally available](https://github.blog/changelog/2023-07-12-pull-request-merge-queue-is-now-generally-available/)

**Comment by megglos** [comment]
> FYI, `bors` project is officially [deprecated](https://bors.tech/newsletter/2023/05/01/tmib-76/). And Merge Queue is now [generally available](https://github.blog/changelog/2023-07-12-pull-request-merge-queue-is-now-generally-available/)

Thanks for raising @nicpuppa ! that puts the topic into a different light then :D Will pull into inbox once more.

**Comment by korthout** [comment]
ZPA triage:
- This may be something that infra could drive as they maintain our bors instance
- But it might be a tool used only by Zeebe
- If we've moved over all usages of Bors, we must inform Infra that they can remove their bors instance(s)
- This would be a relatively large issue because it may affect multiple workflows, requires learning about merge queues, and adjust documentation
- We could explore how to use merge queues on a smaller repo first (like zeebe-analytics or zeebe-cluster-testbench), which may be easier to apply, and we already gain learnings from it
- We still feel that this has priority `later`

**Comment by npepinpe** [comment]
From past experience, there shouldn't be any blockers, as we always merge release changes back into main/stable via PRs. It should be fairly easy to enable this.

However, our focus this month should be on the upcoming release, so we would postpone this to after the release.

**Comment by npepinpe** [comment]
I had a quick look at it during medic week. It seems very straight forward. You simply create a new rule for the target branch, and use the `Test summary` required job at least. Afterwards, it just works.

I did not test with dependabot and so on. We could start with one of the stable branches to minimize impact, and also to allow merging to stable and main in parallel. Once we're happy with it on stable, we can add it to main.

**Comment by npepinpe** [comment]
See: https://github.com/camunda/zeebe/pull/14229
And the merge rule: https://github.com/camunda/zeebe/settings/branch_protection_rules/41406745

**Comment by nicpuppa** [comment]
@npepinpe do you think we can close this issue ? wdyt ?

**Comment by npepinpe** [comment]
Yes, it's pretty much done now. Ole will announce it later.

**Comment by korthout** [comment]
@npepinpe Please ensure the `CONTRIBUTING` guide is updated accordingly. It still mentions using `bors merge` in multiple places. It would also be good to document how contributors can merge their PRs after receiving an approval
